lib/std/process.zig

```diff
@@ -1163,37 +1163,6 @@ pub fn totalSystemMemory() TotalSystemMemoryError!usize {
         .linux => {
             return totalSystemMemoryLinux() catch return error.UnknownTotalSystemMemory;
         },
-        .freebsd, .netbsd, .dragonfly, .macos => {
-            var physmem: c_ulong = undefined;
-            var len: usize = @sizeOf(c_ulong);
-            const name = switch (builtin.os.tag) {
-                .macos => "hw.memsize",
-                .netbsd => "hw.physmem64",
-                else => "hw.physmem",
-            };
-            os.sysctlbynameZ(name, &physmem, &len, null, 0) catch |err| switch (err) {
-                error.NameTooLong, error.UnknownName => unreachable,
-                else => return error.UnknownTotalSystemMemory,
-            };
-            return @as(usize, @intCast(physmem));
-        },
-        .openbsd => {
```

Comment:
the implementation for OpenBSD makes sense.

Reply:
incorrectly reverted by @devnexen; not reverted in #16636